### DISCUSSION TITLE
getting_started/step_by_step/scripting_languages

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#73] [User-contributed notes]
# getting_started/step_by_step/scripting_languages

This lesson will give you an overview of the available scripting languages in Godot. You will learn the pros and cons of each option. In the next part, you will write your first script using GDScri...

https://docs.godotengine.org/en/stable/getting_started/step_by_step/scripting_languages.html

**Comment by typecasto** [comment]
That last screenshot looks really clean, what editor is that?

**Reply by Calinou** [reply]
It looks like a command line text editor (Vim/Emacs). It seems a bit similar to [Doom Emacs](https://github.com/doomemacs/doomemacs) but perhaps it's different.

**Reply by Nodefense** [reply]
Emacs is not a command line text editor.

**Reply by mantisea** [reply]
Emacs is a command line text editor, it's usually opened in a terminal, and yes, it does have a graphical app, but so does Neovim. Also, I think the screenshot is a Neovim config.

**Reply by JMVRy** [reply]
> Emacs is not a command line text editor.

Directly from [the Emacs online manual](https://www.gnu.org/software/emacs/manual/html_node/emacs/Screen.html):
> On a graphical display, such as on GNU/Linux using the X Window System, Emacs occupies a graphical window. On a text terminal, Emacs occupies the entire terminal screen. We will use the term frame to mean a graphical window or terminal screen occupied by Emacs.

They say in a terminal, so it's command-line.

**Reply by kcooper98** [reply]
Definitely an nvim config, maybe AstroNvim with a different theme or their personal config.

**Reply by UberChili** [reply]
> That last screenshot looks really clean, what editor is that?

It is Emacs. Probably DOOM Emacs.


> Emacs is a command line text editor, it's usually opened in a terminal, and yes, it does have a graphical app, but so does Neovim. Also, I think the screenshot is a Neovim config.

It's the other way around. Emacs can be run in the terminal but most every Emacs user uses the GUI.


> They say in a terminal, so it's command-line.

It literally mentions how it can be run as a graphical window or in a terminal screen. But again, it's mostly used as a GUI.

**Reply by JMVRy** [reply]
@UberChili they explicitly said it's not command-line, so I was showing that GNU themselves say it is. I'm not saying it's command-line-only though.

**Reply by nullakan** [reply]
It's DOOM Emacs. In the rare chance that it's not DOOM Emacs, it's safe to say that it's the editor of someone who misses DOOM Emacs very much.
